File: src/main/scala/edu/jhu/fcriscu1/als/graphdb/value/UniProtBlastResult.scala
package org.nygenome.als.graphdb.value

import org.apache.commons.csv.CSVRecord

case class UniProtBlastResult(
      sourceUniprotId:String,
      hitUniprotId:String,
      score:Double,
      eValue:String // no direct way to store exponental data
                             ) {

}

object UniProtBlastResult extends ValueTrait {
val columnHeadings:Array[String] = Array("sourceNumber","sourceId", "sourceLength","sourcePercentIdentity",
  "hitId", "hitLength", "hitPercentIdentity", "score", "eValue", "x")

  def parseCSVRecord(record:CSVRecord):UniProtBlastResult = {
    new UniProtBlastResult(
      parseStringOnPipeFunction(record.get("sourceId"))(1),
      record.get("hitId"),  record.get("score").toDouble,
      record.get("eValue")
      )

  }
}

File: src/main/kotlin/org/biodatagraphdb/alsdb/model/UniProtBlastResult.kt
package org.biodatagraphdb.alsdb.model

import org.apache.commons.csv.CSVRecord

data class UniProtBlastResult(
        val sourceUniprotId: String,
        val hitUniprotId: String,
        val score: Double,
        val eValue: String // no direct way to store exponental data
) {


    companion object : AlsdbModel {
        val columnHeadings = arrayOf("sourceNumber", "sourceId", "sourceLength", "sourcePercentIdentity",
                "hitId", "hitLength", "hitPercentIdentity", "score", "eValue", "x")

        fun parseCSVRecord(record: CSVRecord): UniProtBlastResult =
                UniProtBlastResult(
                        parseStringOnPipe(record.get("sourceId"))[0],
                        record.get("hitId"), record.get("score").toDouble(),
                        record.get("eValue")
                )
    }
}